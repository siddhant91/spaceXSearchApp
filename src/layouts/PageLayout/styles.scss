@import '../../styles/main.scss';

.spacex-page-layout {
	box-sizing: border-box;
	@include padding($spacex-spacing-2, $spacex-spacing-2, $spacex-spacing-2, $spacex-spacing-2);
	@media only screen and (min-width: $spacex-breakpoint-x-large) {
		max-width: 1440px;
		margin: 0 auto;
	}
	&__header {
		position: fixed;
		top: 0;
		background-color: $spacex-background-grey;
		z-index: 1;
		height: $spacex-spacing-11;
		@media only screen and (min-width: $spacex-breakpoint-tablet) {
			height: $spacex-spacing-8;
		}
		&--tile {
			@include margin($spacex-spacing-2, 0, $spacex-spacing-2, 0);
		}
	}
	&__footer {
		position: sticky;
		bottom: 0;
		height: $spacex-spacing-5;
		background-color: $spacex-background-grey;
	}
	&__content {
		@include margin($spacex-spacing-11);
		@media only screen and (min-width: $spacex-breakpoint-tablet) {
			@include margin($spacex-spacing-8);
		}
	}
}
