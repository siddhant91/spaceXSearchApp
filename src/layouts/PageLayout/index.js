/**
 * Page Layout Component which renders the entire page  with header,footer and the main Content
 */
import clsx from 'clsx';
import { string, node } from 'prop-types';

// Styles
import './styles.scss';

const PageLayout = ({ additonalClasses, title, devName, children, ...rest }) => {
	return (
		<div className={clsx('spacex-page-layout', additonalClasses)} {...rest}>
			<header className="spacex-page-layout__header d-flex align-items-center w-100">
				<h1 className="spacex-page-layout__header--title">{title}</h1>
			</header>
			<main className="spacex-page-layout__content">{children}</main>
			<footer className="spacex-page-layout__footer w-100 d-flex justify-content-center align-items-center">
				<strong>Developed by:</strong> {devName}
			</footer>
		</div>
	);
};

PageLayout.propTypes = {
	additonalClasses: string,
	children: node,
	title: string.isRequired,
	devName: string.isRequired,
};

PageLayout.defaultProps = {
	additonalClasses: '',
	children: null,
};

export default PageLayout;
