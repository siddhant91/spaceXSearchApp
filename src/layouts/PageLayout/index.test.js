import { render, cleanup } from '@testing-library/react';

import PageLayout from './index';

afterEach(cleanup);

const renderComponent = () => {
	return (
		<PageLayout title="Some title" devName="Some Dev Name">
			<p>this is child node text</p>
		</PageLayout>
	);
};

it('Matches snapshot', () => {
	const { asFragment } = render(renderComponent());
	expect(asFragment()).toMatchSnapshot();
});

it('renders title', async () => {
	const { findByText } = render(renderComponent());
	expect(await findByText('Some title')).toBeInTheDocument();
});

it('renders footer', async () => {
	const { findByText } = render(renderComponent());
	expect(await findByText('Some Dev Name')).toBeInTheDocument();
});

it('Renders Chidren', async () => {
	const { findByText } = render(renderComponent());
	expect(await findByText('this is child node text')).toBeInTheDocument();
});
