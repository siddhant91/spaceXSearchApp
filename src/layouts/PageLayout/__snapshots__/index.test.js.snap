// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="spacex-page-layout"
  >
    <header
      class="spacex-page-layout__header d-flex align-items-center w-100"
    >
      <h1
        class="spacex-page-layout__header--title"
      >
        Some title
      </h1>
    </header>
    <main
      class="spacex-page-layout__content"
    >
      <p>
        this is child node text
      </p>
    </main>
    <footer
      class="spacex-page-layout__footer w-100 d-flex justify-content-center align-items-center"
    >
      <strong>
        Developed by:
      </strong>
       Some Dev Name
    </footer>
  </div>
</DocumentFragment>
`;
